pandas/tests/arrays/sparse/test_accessor.py

```diff
@@ -212,7 +217,15 @@ def test_series_from_coo(self, dtype, dense_index):
 
         A = scipy.sparse.eye(3, format="coo", dtype=dtype)
         result = pd.Series.sparse.from_coo(A, dense_index=dense_index)
-        index = pd.MultiIndex.from_tuples([(0, 0), (1, 1), (2, 2)])
+
+        index_dtype = np.int64 if dense_index else np.int32
```

Comment:
is it clear we _want_ these to be different?  if not, can you add a TODO comment to that effect

Reply:
I agree, I think they should be the same, but we need to decide which dtype should be used.

We have for *both* 32-bit and 64-bit systems:

```python
>>> import scipy.sparse
>>> A = scipy.sparse.eye(3, format="coo", dtype=dtype)
>>> A.row, A.col
(array([0, 1, 2], dtype=int32), array([0, 1, 2], dtype=int32))
```

I.e. scipy.sparse always uses int32 for index, also on 64-bit systems. The pandas way OTOH is to default to 64-bit dtypes, if an index dtype hasn't been specified, also on 32-bit systems.

So the question is if we should use the pandas or the scipy convention for index dtype when calling `pd.Series.sparse.from_coo`.

My intuition is to follow the scipy way is this instance (i.e. use 32-bit), as `pd.Series.sparse.from_coo` is about converting an existing scipy data structure, so we should follow that data structure's conventions, where possible. What do you think?

Reply:
id add a TODO comment saying roughly what you said here and then punt on it